### ISSUE TITLE
mri_binarize --replace option chains in 7.2.0

### ISSUE THREAD

**Issue by chrisadamsonmcri** [freesurfer/freesurfer#898]
If I run mri_binarize in Freesurfer 7.2.0 to replace labels and I do this

mri_binarize --replace A B --replace B C

A is replaced by C in the output.
That is not the way it used to behave and is counter-intuitive.

**Comment by ahoopes** [comment]
@ayendiki was this an intended effect for the most recent mri_binarize changes?

**Comment by dngreve** [comment]
That seems like a reasonable way to behave. How did it used to behave?

**Comment by chrisadamsonmcri** [comment]
If I ran this

mri_binarize --replace A B --replace B C

Label A would become B in the output.
Label B would become C in the output.

I noticed a new option --replaceonly which seems to be the option that triggers the chaining behaviour. i.e. it executes the block starting at Line 517, which is the chaining version, compared to the one at Line 528, which is the old behaviour.

However, the variable replace_only gets set at Line 1302:
    if(nReplace > 0 || nReplaceNN > 0)
      replace_only = 1;
 Regardless of whether replace or replaceonly is used.

**Comment by buildqa** [comment]
If this is still an issue, then please post this to the freesurfer help list, freesurfer@nmr.mgh.harvard.edu, as more users/developers are likely to see it there.

**Comment by buildqa** [comment]
You can post to the freesurfer list if this still needs attention, freesurfer@nmr.mgh.harvard.edu

**Comment by nschmans** [comment]
I have encountered the problem that [chrisadamsonmcri](https://github.com/chrisadamsonmcri) describes in this issue 898.  The last time I used mri_binarize was using FS v6, where it behaves like this:

mri_binarize --replace A B --replace B C

Label A would become B in the output.
Label B would become C in the output.

here's an example:

--replace 11  15  # map "4th Ventricle" -> "4th-Ventricle"\                                                                                                                                                                                        
--replace 15  72  # map "5th Ventricle" -> "5th-Ventricle"\                                                                                                                                                                                        

i want to convert a seg where the LUT maps 11 as 4th ventricle and 15 as 5th ventricle.  i want to convert these to the FS LUT (using the replacements shown, 15 and 72).  But this results in 4th ventricle getting labeled as 5th ventricle!  Four years ago i ran a script performing 230 replace functions to map Neuromorphometrics labels to FS, and it worked fine (evidenced by Dice testing and inspection).  Today i ran the same script on the same data and get funky results on a half dozen or so structures that follow the example replacement scheme.  My current workaround is to use mri_binarize from v6.
